addons/website_sale/controllers/main.py

```diff
@@ -728,6 +728,10 @@ def address(self, **kw):
                     if not kw.get('use_same'):
                         kw['callback'] = kw.get('callback') or \
                             (not order.only_services and (mode[0] == 'edit' and '/shop/checkout' or '/shop/address'))
+                    # We need to update the pricelist(by the one selected by the customer), because onchange_partner reset it
+                    # We only need to update the pricelist when it is not redirected to /confirm_order
+                    if kw.get('callback'):
```

Comment:
Since `kw` may be populated by anything really, it makes more sense to me to compare it directly to the route we want to 'filter' out.
It also makes it more explicit 🙂 
```suggestion
                    if kw.get('callback', '') != '/shop/confirm_order':
```